src/Voron/StorageEnvironment.cs

```diff
@@ -174,33 +178,50 @@ public StorageEnvironment(StorageEnvironmentOptions options)
                 throw;
             }
         }
-        private async Task IdleFlushTimer()
+
+        ~StorageEnvironment()
+        {
+            if (_log.IsOperationsEnabled)
+                _log.Operations($"Finalizer of storage environment was called. Env: {this}");
+
+            Dispose();
+        }
+
+        private static async Task IdleFlushTimer(EnvironmentWeakReference weakRef)
         {
             try
             {
-                var cancellationToken = _cancellationTokenSource.Token;
-
-                while (cancellationToken.IsCancellationRequested == false)
+                while (true)
```

Comment:
Better, to extract the inner here for a method, and then do the wait outside.

Reply:
Not sure that I understand what you mean. Can you elaborate?